<launch>
    <!-- Detection configuration -->
    <arg name="weights"               default="yolov5s.pt"/>
    <arg name="data"                  default="$(find rsyv5)/scripts/data/coco128.yaml"/>
    <arg name="confidence_threshold"  default="0.5"/>
    <arg name="iou_threshold"         default="0.45"/>
    <arg name="maximum_detections"    default="1000"/>
    <arg name="device"                default="0"/>
    <arg name="agnostic_nms"          default="true"/>
    <arg name="line_thickness"        default="3"/>
    <arg name="dnn"                   default="true"/>
    <arg name="half"                  default="false"/>
    
    
    <!-- Visualize using OpenCV window -->
    <arg name="view_image"            default="true"/>

    <!-- ROS topics -->
    <arg name="input_image_topic"       default="camera/image_raw"/>
    <arg name="output_topic"            default="/yolov5/detections"/>

    <!-- Optional topic (publishing annotated image) -->
    <arg name="publish_image"           default="true"/>
    <arg name="output_image_topic"      default="/yolov5/image_out"/>


    <node pkg="yolov5_ros" name="ros_recognition_yolo" type="ros_recognition_yolo.py" output="screen">
        <param name="weights"               value="yolov5s.pt"/>
        <param name="data"                  value="$(arg data)"/>
        <param name="confidence_threshold"  value="$(arg confidence_threshold)"/>
        <param name="iou_threshold"         value="$(arg iou_threshold)" />
        <param name="maximum_detections"    value="$(arg maximum_detections)"/>
        <param name="device"                value="$(arg device)" />
        <param name="agnostic_nms"          value="$(arg agnostic_nms)" />
        <param name="line_thickness"        value="$(arg line_thickness)"/>
      

        <param name="input_image_topic"     value="$(arg input_image_topic)"/>
        <param name="output_topic"          value="$(arg output_topic)"/>

        <param name="view_image"            value="$(arg view_image)"/>

        <param name="publish_image"         value="$(arg publish_image)"/>
        <param name="output_image_topic"    value="$(arg output_image_topic)"/>
    </node>

</launch>